bread/views/userprofile.py

```diff
@@ -0,0 +1,268 @@
+                                    ),
+                                    widgetattributes={
+                                        "checked": hg.C(
+                                            f"request.session.{layout.DEVMODE_KEY}"
+                                        ),
+                                    },
+                                    onclick=f"fetch('{reverse('devmode', kwargs={'enable': not self.request.session.get(layout.DEVMODE_KEY, False)})}').then((resp) => {{}}).then(() => location.reload(true)); return false;",
+                                ),
+                                style="align-self: flex-end",
+                            ),
+                            layout.DevModeOnly("ok"),
+                            width=7,
+                            style="display: flex; justify-content: flex-end",
+                        ),
+                    ),
+                ),
+                _class="bx--tile",
+            ),
+        )
+
+
+class EditPersonalDataView(EditView):
+    model = get_user_model()
+    fields = ["first_name", "last_name"]
+    urlparams = {}
+
+    def get_object(self):
+        return self.request.user
+
+
+class EditLoginView(EditView):
+    model = get_user_model()
+    fields = ["username", "email"]
+    urlparams = {}
+
+    def get_form_class(self, *args, **kwargs):
+        class EditLoginForm(super().get_form_class(*args, **kwargs)):
+
+            current_password = forms.CharField(
+                label=_("Confirm changes with current password"),
+                strip=False,
+                widget=forms.PasswordInput(),
+                help_text=_(
+                    "Please enter your current password to confirm these changes"
+                ),
+            )
+
+            def clean_current_password(self_inner):
+                current_password = self_inner.cleaned_data["current_password"]
+                if not self.request.user.check_password(current_password):
+                    raise ValidationError(
+                        _(
+                            "Your current password was entered incorrectly. Please enter it again."
+                        ),
+                    )
+                return current_password
+
+        return EditLoginForm
+
+    def get_form(self, *args, **kwargs):
+        ret = super().get_form(*args, **kwargs)
+        return ret
+
+    def get_layout(self):
+        ret = super().get_layout()
+        ret.append(layout.form.FormField("current_password"))
+        return ret
+
+    def get_object(self):
+        return self.request.user
+
+
+class EditPermissionsView(EditView):
+    model = get_user_model()
+    fields = ["is_active", "is_superuser", "is_staff"]
+    urlparams = {}
+
+    def check_permissions(self, request):
+        if not request.user.is_superuser:
+            return HttpResponseForbidden()
+
+    def get_object(self):
+        return self.request.user
+
+
+def profile_field(fieldname, password=False):
```

Comment:
I would create two functions "profile_regular_field(fieldname)" and "profile_password_field(fieldname)" which both call a function "profile_field(fieldname, fieldvalue)".
For me, a function should focus as much as possible to do one thing instead of using if-else.

Reply:
Agreed, this is unclean.